File: city.rb
class City
  def initialize
    @food = 500000
    @people = []
    10.times { @people << Person.new(rand(15..35)) }
  end

  def turn
    @food -= @people.count * 12 # people need to eat
    @people.each do |person|
      person.get_older!
    end

    inspect
  end

  def inspect
    a = @people.map do |person|
      "#{person.age}#{person.gender[0]}"
    end
    p a
    p "Remaining food: #{@food}"
  end
end

class Person
  GENDERS = [:male, :female]

  attr_reader :age, :gender

  def initialize(age = 0)
    @age = age
    @gender = GENDERS.sample
  end

  def get_older!
    @age += 1
  end

  def male?
    @gender == :male
  end

  def female?
    @gender == :female
  end
end

File: src/city.js
import { Person } from './person';
import { randomBetween } from './utils/random';

export class City {
  constructor(initialPeople = 10) {
    this.food = 500000;
    this.people = []
    for(var i = 0; i < initialPeople; i++) {
      this.people.push(new Person(randomBetween(15, 25)));
    }
  }

  turn() {
    this.food -= this.people.length * 12; // people eat each month
    this.people.forEach(p => p.turn());
  }

  inspect() {
    let peopleInspect = this.people.map(p => `${p.inspect()}`)
    return `${peopleInspect}; Remaining food: ${this.food}`;
  }
}
